src/main/java/org/radarcns/RestructureAvroRecords.java

```diff
@@ -70,27 +72,36 @@
 
     private long processedFileCount;
     private long processedRecordsCount;
-    private static final boolean USE_GZIP = "gzip".equalsIgnoreCase(System.getProperty("org.radarcns.compression"));
+    private static boolean USE_GZIP;
+    private static boolean DO_DEDUPLICATE;
 
-    // Default set to false because causes loss of records from Biovotion data. https://github.com/RADAR-base/Restructure-HDFS-topic/issues/16
-    private static final boolean DO_DEDUPLICATE = "true".equalsIgnoreCase(System.getProperty("org.radarcns.deduplicate", "false"));
+    private static final CommandLineArgs commandLineArgs = new CommandLineArgs();
+    private static final JCommander parser = JCommander.newBuilder().addObject(commandLineArgs).build();
 
     public static void main(String [] args) throws Exception {
-        if (args.length != 3) {
-            System.out.println("Usage: hadoop jar restructurehdfs-all-0.2.jar <webhdfs_url> <hdfs_root_directory> <output_folder>");
-            System.exit(1);
+
+        parser.setProgramName("hadoop jar restructurehdfs-all-0.3.3.jar");
+        parser.parse(args);
+
+        if(commandLineArgs.help) {
+            parser.usage();
+            System.exit(0);
         }
 
+        USE_GZIP = "gzip".equalsIgnoreCase(commandLineArgs.compression);
+        DO_DEDUPLICATE = commandLineArgs.deduplicate;
```

Comment:
If these parameters are explicitly parsed anyway, perhaps just pass them to the class instead of having them static?

Reply:
In that case, the restructureRecords could possibly take a builder pattern.